lib/geoengineer/resource.rb

```diff
@@ -225,19 +225,23 @@ def setup_tags_if_needed
     tags {} unless tags
   end
 
-  def merge_project_tags
-    return unless self.project && self.project.tags && self.support_tags?
+  def merge_parent_tags
+    sources = %i(project environment)
+    sources.each do |source|
+      next unless self.support_tags?
+      next unless self.send(source)&.tags
 
-    setup_tags_if_needed
+      setup_tags_if_needed
 
-    self
-      .project
-      .all_tags
-      .map(&:attributes)
-      .reduce({}, :merge)
-      .each { |key, value| tags.attributes[key] ||= value }
+      self
```

Comment:
Bit odd to use `self` here unless it's part of our coding style. `self` should only be explicitly required if you were trying to set an attribute of a class (instead of setting a local variable).

Not a blocker in any case; you didn't add this.